### PR TITLE
Add the SLP6 Release Note

### PR DESCRIPTION
**Pull request by praneesha**
## Purpose
Add the SLP6 release note.

> Fixes #

## Check List

- [ ] **Page Addition**
  - [ ] Add `permalink` to pages
  - [ ] If contains empty folder(s), Add front-matter `redirect_to:`

- [ ] **Page Rename**
  - [ ] Add front-matter `redirect_from`
  - [ ] Add front-matter `redirect_to:` (If applicable)

### CODE REVIEW

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 1]
```diff
@@ -0,0 +1,354 @@
+---
+layout: ballerina-blank-page
+title: Release Note
+---
+### Overview of Ballerina Swan Lake Preview 6 
+
+This release is the sixth preview version of Ballerina Swan Lake. This release includes a new set of language features along with improvements and bug fixes to the compiler, runtime, standard libraries, and developer tooling.
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:7]:
```suggestion
This release is the sixth preview version of Ballerina Swan Lake. This release includes a new set of language features along with improvements and bug fixes to the compiler, runtime, standard library, and developer tooling.
```

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+| io.ballerina.runtime.api.types and  io.ballerina.runtime.api.values | Holds the Ballerina runtime API types and values.                                                                                                                                                                                                                             |
+| io.ballerina.runtime.api.Environment                                | Interop functions can have the `Environment` class variable as the first parameter.  This provides APIs to get the current module, strand ID, strand name, and strand metadata.  Also, with the environment instance, you can set and get the strand local properties. |
+| io.ballerina.runtime.api.Runtime                                    | Environment also provides an instance of current runtime class which will contain apis to invoke ballerina object functions.                                                                                                                                              |
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:258]:
```suggestion
| io.ballerina.runtime.api.Runtime                                    | Environment also provides an instance of the current runtime class which will contain apis to invoke ballerina object functions.                                                                                                                                              |
```

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+        import ballerina/http;
+        import ballerina/c2c as _;
+
+        service hello on new http:Listener(9090) {
+            resource function sayHello(http:Caller caller, http:Request req) returns error? {
+                check caller->respond("Hello, World!");
+            }
+        }
+
+        ```
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:335]: The above code block is not properly displayed.

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+#### Breaking Changes
+
+1. It is no longer possible to specify the version in an import declaration. A specific version can be imported by specifying the package dependency along with the version in the `Ballerina.toml` file.
+
+        ```toml
+        [[dependency]]
+        org = "ballerina"
+        name = "stringutils"
+        version = "0.5.2"
+        ```
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:348]: The above code block is not properly displayed.

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
+#### Runtime
+
+##### Support for DynamicListener
+
+ A dynamic listener can be registered and deregistered at the runtime as follows.
+
+ ```ballerina
+ ModListener listenerVar = new(); // Where ModListener is a DynamicListener 
+	runtime:registerListener(listenerVar);
+	runtime:deregisterListener(listenerVar);
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:243]: Incorrect indentation for the 2nd and 3rd lines

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+
+        service hello on new http:Listener(9090) {
+            resource function sayHello(http:Caller caller, http:Request req) returns error? {
+                check caller->respond("Hello, World!");
+            }
+        }
+
+        ```
+
+2. Removed the Kubernetes module and replaced it by c2c.
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:337]:
```suggestion
2. Removed the Kubernetes module and replaced it with `c2c`.
```

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 7]
```diff
@@ -0,0 +1,354 @@
...
+        ```toml
+        [[dependency]]
+        org = "ballerina"
+        name = "stringutils"
+        version = "0.5.2"
+        ```
+
+2. As the initial step of upcoming changes for the error detail, the detail-type descriptor of an error type descriptor can no longer be a closed record.
+
+3. The Kubernetes module is removed and replaced by the c2c module.
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:352]:
```suggestion
3. The Kubernetes module is removed and replaced by the `c2c` module.
```

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+- Ability to use the `isolated` qualifier  with an object-constructor expression to construct an `isolated` object
+- Support to mark an object-type definition as a `distinct` object type 
+- Allow using the object constructor to create distinct objects
+- Support to register and deregister a dynamic listener at the runtime
+- Introduction of the new runtime APIs
+- Revamp of the `File/Filepath` modules
+- Introduction of new high-level APIs to provide I/O operations
+- Introduction of a new code-action `Add Type Cast`
+- Revamp of the code-action extension APIs 
+- Introduction of code to cloud to simplify deploying Ballerina code in the cloud
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:62]:
```suggestion
- Introduction of code to cloud (`c2c`) to simplify deploying Ballerina code in the cloud
```

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 9]
```diff
@@ -0,0 +1,354 @@
...
+booleanVar = false
+```
+
+Currently, Ballerina supports configurable variables of the types `int`, `float`, `boolean`, and `string`. Future versions will add support for configurable variables of any type that is a subtype of `readonly&anydata`.
+
+##### Isolated Variables
+
+Module variable declarations with an initial value expression can now be marked as `isolated`. An `isolated` variable can be accessed only within a `lock` statement.
+
+Similar to isolated objects, isolated variables are also considered as isolated roots, which guarantee that their mutable state is only reachable via the isolated root. As with `isolated` objects, when an `isolated` variable is accessed within a `lock` statement, additional rules apply to how values are transferred in and out of the `lock` statement.
```
Comment (anupama-pathirage) [inline downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:105]:
```suggestion
Similar to `isolated` objects, `isolated` variables are also considered as isolated roots, which guarantee that their mutable state is only reachable via the isolated root. As with `isolated` objects, when an `isolated` variable is accessed within a `lock` statement, additional rules apply to how values are transferred in and out of the `lock` statement.
```

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+
+    - From `filepath` - `extension`, `getPathSeparator`, `getPathListSeparator`, `isReserved`, and `matches`
+    - From `file` - `exists`, `tempDir`
+
+3. The implementation of some APIs has been modified.
+
+    - From `Filepath` - `normalize`
+    - From `File` - `createDir`, `remove`, `copy`, `readDir`, `getMetaData` (earlier `getFileInfo`)
+
+4. Some APIs have been introduced newly.
```
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:279]: You say "newly introduced" only when it is used as an adjective. In this, it should be "introduced newly".

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md (hunk 5 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md]: Fixed.

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md (hunk 3 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:331]: We need 2 tabs to get code snippets displayed properly on the website although that doesn't look proper in this GitHub view.

They will be fine once they are rendered on the website.

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md (hunk 4 above)
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:344]: We need 2 tabs to get code snippets displayed properly on the website although that doesn't look proper in this GitHub view.

They will be fine once they are rendered on the website.

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 11]
```diff
@@ -0,0 +1,354 @@
...
+        org = "ballerina"
+        name = "stringutils"
+        version = "0.5.2"
+        ```
+
+2. As the initial step of upcoming changes for the error detail, the detail-type descriptor of an error type descriptor can no longer be a closed record.
+
+3. The Kubernetes module is removed and replaced by the c2c module.
+
+
```
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md]: I don't see these new lines anymore. @MaryamZi - Am I missing something?

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 12]
```diff
@@ -0,0 +1,354 @@
...
+1. It is no longer possible to specify the version in an import declaration. A specific version can be imported by specifying the package dependency along with the version in the `Ballerina.toml` file.
+
+        ```toml
+        [[dependency]]
+        org = "ballerina"
+        name = "stringutils"
+        version = "0.5.2"
+        ```
+
+2. As the initial step of upcoming changes for the error detail, the detail-type descriptor of an error type descriptor can no longer be a closed record.
```
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md]: +1

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 13]
```diff
@@ -0,0 +1,354 @@
...
+       self.r = r;
+       self.color = color;
+   }
+}
+
+Circle c = new ColoredCircle(0, 0, 0, [123, 100, 100]);
+Circle d = new ColoredCircleLookAlike(0, 0, 0, [123, 100, 100]); // Not allowed.
+```
+
+The `ColoredCircle` is a subtype of the `Circle` distinct object type even though `ColoredCircle` is not marked as a distinct class.
```
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md]: @MaryamZi - To whom should the comment be addressed?

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 14]
```diff
@@ -0,0 +1,354 @@
...
+
+###### Distinct Object Constructor
+
+If no type reference is provided for the object constructor, contextually, the expected type must be definite and type-ids of the constructed object are the type-ids of the contextually-expected type. This allows using the object constructor to create distinct objects.
+
+#### Runtime
+
+##### Support for DynamicListener
+
+ A dynamic listener can be registered and deregistered at the runtime as follows.
```
[the remark this answers is not in the document]
Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:234]: +1

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md (hunk 1 above)
Comment (anupama-pathirage) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:7]: Why this was resolved without committing the suggestion?

Comment (praneesha) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:7]: I committed this but seems like something wrong has happened. Sorry, about it.

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 15]
```diff
@@ -0,0 +1,354 @@
...
+
+    isolated object {} isolatedOb = ob; // Now valid.
+}
+```
+
+##### Support for Distinct Objects
+
+###### Distinct Object Type
+
+Marking an object type definition as a `distinct` object type is supported now.
```
[the remark this answers is not in the document]
Comment (rdhananjaya) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md]: +1

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (MaryamZi) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md]: @rdhananjaya

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (MaryamZi) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md:287]: Shall we say 
```suggestion
4. Some new APIs have been introduced.
```
then?

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (MaryamZi) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md]: Seems like this has been fixed.

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md (hunk 13 above)
[the remark this answers is not in the document]
Comment (rdhananjaya) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md]: +1 anyway this is already there https://github.com/ballerina-platform/ballerina-dev-website/pull/1534/files#diff-cb36d247980a51a9428eaa3478638c30a950c569b29a96a645f1d6fa1a97152cR222

**Review by NipunaRanasinghe — COMMENTED** [review]
@praneesha Let's update the release version as `preview 7`, as we discussed.

File: downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md [hunk 16]
```diff
@@ -0,0 +1,354 @@
...
+
+All the Ballerina runtime internal exceptions will not be exposed. The `BError` should be used to handle Ballerina runtime errors. The `io.ballerina.runtime.api.creators.ErrorCreator` provides the required APIs to create runtime errors.                                                                         
+
+#### Standard Library
+
+##### Revamp File and Filepath Modules
+
+The `File` and `Filepath` APIs have been revamped in this release. A summarized list of the changes done is as follows.
+
+1. The `File` and `Filepath` modules have been merged together. There will be no `Filepath` module anymore.
```
[the remark this answers is not in the document]
Comment (ThishaniLucas) [reply downloads/swan-lake-release-notes/swan-lake-preview6/RELEASE_NOTE.md]: Fixed.

**Comment by praneesha** [comment]
> @praneesha Let's update the release version as `preview 7`, as we discussed.

Done.

**Review by NipunaRanasinghe — APPROVED** [review]

**Pull request merged**